### PR TITLE
Refactoring #3. Fixed functionality for longest streak and current st…

### PR DESCRIPTION
**Pull request by MrAlexDeluxe**
Refactoring #3. Fixed functionality for longest streak and current streak. Problem was that array was not sorted, but i assumed that it is.

### CODE REVIEW

File: zeeguu/templates/activity_graph/activity_graph.html [hunk 1]
```diff
@@ -179,25 +177,42 @@
...
+        var extracted_month = str_parts[1] - 1;
         var extracted_year = str_parts[0];
 
         return new Date(extracted_year, extracted_month, extracted_day);
     }
 
-    function longest_streak() {
-        var max = 0, temp = 0;
-        var prev_date = bookmark_counts_by_date_array[0].date;
+    function compare_dates(date1, date2) {
```
Comment (mircealungu) [inline zeeguu/templates/activity_graph/activity_graph.html]: `compare_date_strings` would be a more intention revealing name :)

File: zeeguu/templates/activity_graph/activity_graph.html [hunk 2]
```diff
@@ -179,25 +177,42 @@
...
+    }
 
-        bookmark_counts_by_date_array.forEach(function (entry) {
-            if (compute_distance_between_dates_in_days(str_to_date(entry.date), str_to_date(prev_date)) <= 1) {
-                temp++;
-                prev_date = entry.date;
-            } else {
+    bookmark_counts_by_date_array.sort(compare_dates); // sort array based on the dates
+
+    function longest_streak() {
```
Comment (mircealungu) [inline zeeguu/templates/activity_graph/activity_graph.html]: if you'll want to reuse this function for exercises, you have to transform the `bookmark_counts_by_date_array` into a parameter...

**Comment by mircealungu** [comment]
Aha :) Oki, this makes sense and I'm sure it's fixed now. However, there's a few small cosmetic changes that still need to be taken care of.

File: zeeguu/templates/activity_graph/activity_graph.html (hunk 1 above)
Comment (MrAlexDeluxe) [inline zeeguu/templates/activity_graph/activity_graph.html]: wrong, I am comparing only dates, not a bookmarks.

Comment (mircealungu) [inline zeeguu/templates/activity_graph/activity_graph.html]: you sure?
if I remember the code, it was saying date1.date and date2.date.
how can a date have the .date field?

M.

Comment (MrAlexDeluxe) [inline zeeguu/templates/activity_graph/activity_graph.html]: true, I am extracting date from the array element, but also it is not a bookmark_count either

Comment (mircealungu) [inline zeeguu/templates/activity_graph/activity_graph.html]: hmm. but what's that array element? it's not a date.
i thought it's a dict that contains a date and a count!
case in which i thought about calling it a bookmark count element... but
sure, if you have a better name, pls go ahead :)

Comment (MrAlexDeluxe) [inline zeeguu/templates/activity_graph/activity_graph.html]: I can agree with this. 
compare_date_strings(array_element1, array_element2) {...}

Comment (MrAlexDeluxe) [inline zeeguu/templates/activity_graph/activity_graph.html]: array element can be any element from any array, which contains at least (.date) field. 
this function will be also reused later for other arrays, which will have a date field, but will have something else than bookmark_count, like exerice_count

Comment (mircealungu) [inline zeeguu/templates/activity_graph/activity_graph.html]: I can agree with that :)

Comment (mircealungu) [inline zeeguu/templates/activity_graph/activity_graph.html]: is the array element a dictionary, which has a `date` key?

Comment (MrAlexDeluxe) [inline zeeguu/templates/activity_graph/activity_graph.html]: yes, in this case it is dictionary, which contains at least a `date` key.

Comment (mircealungu) [inline zeeguu/templates/activity_graph/activity_graph.html]: excellent, then the name of the parameters can be `dict1` and `dict2` ;)
and in a comment you should specify that it expects a date string as one of
the keys! pair programming is what we do,  isn't it? :)

Comment (MrAlexDeluxe) [inline zeeguu/templates/activity_graph/activity_graph.html]: yes. :)

Comment (mircealungu) [inline zeeguu/templates/activity_graph/activity_graph.html]: Alex, can you push your latest changes?
Thx,
M.

**Pull request merged**